engine/src/main/java/org/terasology/rendering/opengl/GLSLShader.java

```diff
@@ -322,85 +328,68 @@ private void compileShader(int type, Set<ShaderProgramFeature> features) {
             shader.append(shaderProgramBase.getVertexProgram());
         }
 
+        return shader.toString();
+    }
+
+    private void dumpCode(int type, Set<ShaderProgramFeature> features, String sourceCode) {
         String debugShaderType = "UNKNOWN";
         int featureHash = ShaderProgramFeature.getBitset(features);
         if (type == GL20.GL_FRAGMENT_SHADER) {
-            fragmentPrograms.put(featureHash, shaderId);
             debugShaderType = "FRAGMENT";
         } else if (type == GL20.GL_VERTEX_SHADER) {
-            vertexPrograms.put(featureHash, shaderId);
             debugShaderType = "VERTEX";
         }
 
-        Config config = CoreRegistry.get(Config.class);
-        if (config.getRendering().isDumpShaders()) {
-            // Dump all final shader sources to the log directory
-            final String strippedTitle = getURI().toString().replace(":", "-");
-
-            String fname = debugShaderType.toLowerCase() + "_" + strippedTitle + "_" + featureHash + ".glsl";
-            Path path = PathManager.getInstance().getShaderLogPath().resolve(fname);
-            try (BufferedWriter writer = Files.newBufferedWriter(path, TerasologyConstants.CHARSET)) {
-                writer.write(shader.toString());
-            } catch (IOException e) {
-                logger.error("Failed to dump shader source.");
-            }
+        // Dump all final shader sources to the log directory
+        final String strippedTitle = getURI().toString().replace(":", "-");
+
+        String fname = debugShaderType.toLowerCase() + "_" + strippedTitle + "_" + featureHash + ".glsl";
+        Path path = PathManager.getInstance().getShaderLogPath().resolve(fname);
+        try (BufferedWriter writer = Files.newBufferedWriter(path, TerasologyConstants.CHARSET)) {
+            writer.write(sourceCode);
+        } catch (IOException e) {
+            logger.error("Failed to dump shader source.");
         }
+    }
 
-        GL20.glShaderSource(shaderId, shader.toString());
-        GL20.glCompileShader(shaderId);
+    private int compileShader(int type, Set<ShaderProgramFeature> features) {
+        int shaderId = GL20.glCreateShader(type);
 
-        StringBuilder error = new StringBuilder();
-        boolean success = printLogInfo(shaderId, error);
+        String shader = createShader(type, features);
 
-        String errorLine = "";
-        if (error.length() > 0) {
-            try {
-                Pattern p = Pattern.compile("-?\\d+");
-                Matcher m = p.matcher(error.toString());
+        Config config = CoreRegistry.get(Config.class);
```

Comment:
The renderingConfig and the renderingDebugConfig are obtained and reobtained from the CoreRegistry throughout the class. Can we get them only once, into a couple of variables and use those?